### ISSUE TITLE
Ability to pass vault password file to connection=ansible

### ISSUE THREAD

**Issue by apenney** [philpep/testinfra#97]
Hi,

We're working on using testinfra to test our ansible code but we have a vault file under our group_vars/all directory.  Within ansible this triggers an automatic attempt to decrypt the vault (to parse all group vars to be added to the inventory).

This leads to a backtrace like:

_______________________________________ ERROR collecting test_clustering.py _______________________________________
../../../../venv-ansible/lib/python2.7/site-packages/testinfra/plugin.py:116: in pytest_generate_tests
    ansible_inventory=metafunc.config.option.ansible_inventory,
../../../../venv-ansible/lib/python2.7/site-packages/testinfra/__init__.py:37: in get_backends
    _BACKENDS_CACHE[key] = backend.get_backends(hosts, **kwargs)
../../../../venv-ansible/lib/python2.7/site-packages/testinfra/backend/__init__.py:88: in get_backends
    for name in klass.get_hosts(host, **kw):
../../../../venv-ansible/lib/python2.7/site-packages/testinfra/backend/ansible.py:68: in get_hosts
    return ansible_runner.get_hosts(kwargs.get("ansible_inventory"), host)
../../../../venv-ansible/lib/python2.7/site-packages/testinfra/utils/ansible_runner.py:209: in get_hosts
    return AnsibleRunner(host_list).get_hosts(pattern)
../../../../venv-ansible/lib/python2.7/site-packages/testinfra/utils/ansible_runner.py:147: in __init__
    host_list=host_list,
../../../../venv-ansible/lib/python2.7/site-packages/ansible/inventory/__init__.py:85: in __init__
    self.parse_inventory(host_list)
../../../../venv-ansible/lib/python2.7/site-packages/ansible/inventory/__init__.py:143: in parse_inventory
    group.vars = combine_vars(group.vars, self.get_group_variables(group.name))
../../../../venv-ansible/lib/python2.7/site-packages/ansible/inventory/__init__.py:508: in get_group_variables
    self._vars_per_group[groupname] = self._get_group_variables(groupname, vault_password=vault_password)
../../../../venv-ansible/lib/python2.7/site-packages/ansible/inventory/__init__.py:526: in _get_group_variables
    vars = combine_vars(vars, self.get_group_vars(group))
../../../../venv-ansible/lib/python2.7/site-packages/ansible/inventory/__init__.py:706: in get_group_vars
    return self._get_hostgroup_vars(host=None, group=group, new_pb_basedir=new_pb_basedir)
../../../../venv-ansible/lib/python2.7/site-packages/ansible/inventory/__init__.py:745: in _get_hostgroup_vars
    results = combine_vars(results, self._variable_manager.add_group_vars_file(base_path, self._loader))
../../../../venv-ansible/lib/python2.7/site-packages/ansible/vars/__init__.py:561: in add_group_vars_file
    (name, data) = self._load_inventory_file(path, loader)
../../../../venv-ansible/lib/python2.7/site-packages/ansible/vars/__init__.py:518: in _load_inventory_file
    _found, results = self._load_inventory_file(path=p, loader=loader)
../../../../venv-ansible/lib/python2.7/site-packages/ansible/vars/__init__.py:533: in _load_inventory_file
    data = loader.load_from_file(path)
../../../../venv-ansible/lib/python2.7/site-packages/ansible/parsing/dataloader.py:112: in load_from_file
    (file_data, show_content) = self._get_file_contents(file_name)
../../../../venv-ansible/lib/python2.7/site-packages/ansible/parsing/dataloader.py:170: in _get_file_contents
    data = self._vault.decrypt(data)
../../../../venv-ansible/lib/python2.7/site-packages/ansible/parsing/vault/__init__.py:171: in decrypt
    raise AnsibleError("Decryption failed")

Ideally I could do one of two things:

1) Option to provide an ansible.cfg to be parsed.
2) Ability to pass in vault file directly.

**Comment by philpep** [comment]
Hi,

Ansible connection backend is using ansible python API, so it must behave almost like ansible does (loading ansible.cfg, inventory file, vars files). One thing that testinfra does not is accept ansible options that are passed on the command line (eg: `--vault-password-file`, `--ask-vault-pass` etc), so it's better if they are in ansible.cfg.

Could you tell me more about your setup (content of ansible.cfg especially and how to you call ansible) ? I'll try to reproduce the issue and see what we can do.

**Comment by philpep** [comment]
Also what ansible version do you use ?

**Comment by rabadin** [comment]
I have the same issue.
I believe the problem stems from the way ansible (v1) passes the path of the file containing the vault password: it's done in the `base_parser` method in `ansible/utils/__init__.py`:
```
    parser.add_option('--vault-password-file', default=constants.DEFAULT_VAULT_PASSWORD_FILE,
        dest='vault_password_file', help="vault password file")
```
`constants.DEFAULT_VAULT_PASSWORD_FILE` has the correct value (i.e. it has the path that is specified by `vault_password_file` in ansible.cfg) but the problem, I think, is that this code (`base_parser`) is *not* called when one calls ansible from the Python API.
As a result, ansible is unable to decrypt the vault and fails with the error above.

**Comment by philpep** [comment]
@rabadin could you try calling testinfra with the ``ANSIBLE_VAULT_PASSWORD_FILE=~/your/vault/passfile`` environment variable (or set ``vault_password_file`` in ``ansible.cfg``) ? It may fix your issue I think.

**Comment by rabadin** [comment]
I've tested both (calling testinfra with the ANSIBLE_VAULT_PASSWORD_FILE and setting `vault_password_file` in `ansible.cfg`): same problem.

```
ANSIBLE_VAULT_PASSWORD_FILE=/Users/<me>/.config/ansible/.vault.pass testinfra --sudo --connection=ansible --ansible-inventory /Users/<me>/Dev/Code/<proj>/inventory/dyn_inventory.py /Users/rabadin/Dev/Code/<proj>/tests/integration/test_*.py
```

**Comment by philpep** [comment]
Thanks @rabadin 
I'll try to reproduce and see if we need to add a new option in testinfra to handle this.

**Comment by philpep** [comment]
I've a WIP fix in #112 found a lot of bugs in the ansible backend. Needed to go deep inside ansible internal api due to the lack of an usable public api.
